### ISSUE TITLE
Incorrect induced charge dragging alert when jumping to center or near wall, then moving left

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#341]
@jessegreenberg, awesome work on implementing Induced charge dragging alerts.

One adjustment needed when jumping to center or to near wall with JC and JN hot keys.

In these interactions, it is best to use the Induced charge dragging (generic) pattern. This pattern contains no description for the amount of deflection in order to not confuse descriptions of deflection if further dragging towards or away from the wall continues.

See section D Interact alerts - Jumping
[Jump to Center and Jump to Near Wall](https://docs.google.com/document/d/1WNVq5_-Hc9LSdeZVSn2edlwYwpHtGNT1ThFD9NXpZ5Q/edit#heading=h.rdqv3cnbwecf)

The pattern is like this:
- At [[center]] of Play Area. Negative charges in [[upper wall]] move away from [[Yellow balloon]].
- [[Near wall]]. Negative charges in [[upper wall]] move away from [[Yellow balloon]].

**Comment by jessegreenberg** [comment]
@terracoda, done in the above commit. The result is that when a charged balloon jumps to [near] wall, then to center, it might sound like this:

- [[Near wall]]. Negative charges in [[upper wall]] move away from [[Yellow balloon]].
- At [[center]] of Play Area. Negative charges in [[upper wall]] move away from [[Yellow balloon]].

Even though in the second jump charges returned a bit. Is this OK?

**Comment by terracoda** [comment]
@jessegreenberg, excellent point.
Unless we can keep track of the number of jumps, I think it might be best to think of a jump as a new starting point of dragging interaction.

**Comment by jessegreenberg** [comment]
Hmm, I am not thrilled about counting number of jumps. But one thing we could do is use the same change in induced charge description, so it would sound like

- JN: [[Near wall]]. Negative charges in [[upper wall]] begin to  move away from [[Yellow balloon]].
- JC: At [[center]] of Play Area. Negative charges in [[upper wall]] begin to return.

**Comment by terracoda** [comment]
@jessegreenberg, actually using the "begin to move way" seems like a better starting description, but would that work if user starts off dragging and then jumps? For example:

User is dragging balloon right: 
- On [[upper-right side]] of play area. Negative charges in [[upper wall]] begin to move away from yellow balloon [[a little]].
And then they jump with JN:
- Near [[upper wall]]. Negative charges in [[upper wall]] move away from [[Yellow balloon]] [[a little more]].

Or they start off dragging away from wall and lets say the balloon has a lot of charge:
- Near [[lower wall]]. Negative charges in [[lower wall]] begin to return.
Then they jump to center:
- At [[lower-center]] of Play Area. Negative charges in [[lower wall]] return a little more.

@jessegreenberg, am I correct to assume that it gets tricky if a user is switching randomly between dragging and jumping?

Is there a way to watch for (i.e., take note of) the first induced effect, and then to continue describing the deflection or the return regardless of whether the user jumps with the hot key combinations or drags with the Arrow keys?

**Comment by terracoda** [comment]
@jessegreenberg, I'll have a listen to latest version and get back to yo on this.

The nice thing about the generic "Negative charges in [[wall]]  move away from [[Yellow balloon]]" is that it is always true if there is a positive  induced charge effect. 

The finer details about moving away and returning might just have to be explored during direct interaction.

**Comment by terracoda** [comment]
@jessegreenberg, we'll need some tweaks to these Jump and dragging alerts for induced charge.

I'll put this on my TO Do list.

Things I noticed today (Feb 7) are:
1. Jump to Near wall, and then move left and I hear "return a little more" instead of "begin to return"
2. Jump to wall, hear Induced charge (generic) and then drag left and hear  "return a little more" instead of "begin to return"
3. Jumping to wall should not be Induced charge (generic). The deflection amount should be added. I think it should a full induced description, "At wall. Negative charges in {{wall}} move away from {{Yellow Balloon}} {{quite a lot}}. Positive charges do not move." The "No transfer" string can then be added when rubbing against the wall starts.

**Comment by terracoda** [comment]
@terracoda to re-test. These issues may have been addressed.

**Comment by terracoda** [comment]
Pressing  A key or left arrow key should never trigger: "Negative charges in wall begin to move away from Yellow balloon."

**Comment by terracoda** [comment]
@jessegreenberg, jumping then dragging is working pretty good.

I found 2 issues:
1. When I use JC or JN and then move left, I sometimes hear "Negative charges in wall begin to move away from Yellow balloon." This usually only happens after a single jump first then a move left.

2. One time after JC to center then a move up, I heard, "At center of Play Area. In wall, no change in charges." If you could make it so we only hear "In wall, no change in charges." when the balloon is actually rubbing against the wall, that would be great. I think this description fired when there was a very tiny induced charge effect. I could not re-produce it.

I tested all sorts of other jumping and moving scenarios:
- JW move left - good
- JN - good
- JN then move right - good
- JN, then move right, then move left - good
- JN, then move up,/down, then move left - good
- JC when there is a measurable induced charge effect, then move up - good
- JC when there is a measurable induced charge effect, then move up, then move left - good

**Comment by jessegreenberg** [comment]
There are lots of problems with this because the change in induced charge is driven by position changes to the balloon. It should be based on changing model  forces from the balloon. Ill need to reimplement this.

**Comment by terracoda** [comment]
@jessegreenberg, let me know if need to have a meeting about these jump alerts.

**Comment by jessegreenberg** [comment]
Thanks @terracoda, this one was tough but I think I have a working solution. I just need to clean it up.

**Comment by terracoda** [comment]
@jessegreenberg, awesome!

**Comment by jessegreenberg** [comment]
https://github.com/phetsims/balloons-and-static-electricity/commit/e1536f5766c436a5d5e3048ad311e7ffa34a7e2c was intended to fix this. I noticed an issue still with these alerts but can't remember what was happening in sim. Ill need to revisit.

**Comment by jessegreenberg** [comment]
@terracoda https://github.com/phetsims/balloons-and-static-electricity/issues/341#issuecomment-368938070 is fixed, and various combinations of jumping sound good. Let me know if you notice other issues with this. Closing.
